libs/langchain/langchain/schema/runnable/locals.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+from typing import Any, AsyncIterator, Dict, Iterator, Mapping, Optional, Union
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForChainRun,
+    CallbackManagerForChainRun,
+)
+from langchain.load.serializable import Serializable
+from langchain.schema.runnable.base import Input, Output, Runnable
+from langchain.schema.runnable.config import RunnableConfig
+from langchain.schema.runnable.passthrough import RunnablePassthrough
+
+
+class PutLocalVar(RunnablePassthrough):
+    key: Union[str, Mapping[str, str]]
+    """The key(s) to use for storing the input variable(s) in local state.
+    
+    If a string is provided then the entire input is stored under that key. If a 
+        Mapping is provided, then the map values are gotten from the input and 
+        stored in local state under the map keys.
+    """
+
+    def __init__(self, key: Union[str, Mapping[str, str]], **kwargs: Any) -> None:
+        super().__init__(key=key, **kwargs)
+
+    def _concat_put(
```

Comment:
We don't have any clean way to get PutLocalVariable produce a deep copy of the config that will get passed along right -- since config is not part of the output?

A bit worried that because the deep copy isn't localized here, we may end up bumping into a bug at  some point where we forget to include a deep copy